src/sys/ptrace.rs

```diff
@@ -6,68 +6,87 @@ use libc::{self, c_void, c_long, siginfo_t};
 use ::unistd::Pid;
 use sys::signal::Signal;
 
-pub mod ptrace {
-    use libc::c_int;
-
-    cfg_if! {
-        if #[cfg(any(all(target_os = "linux", arch = "s390x"),
-                     all(target_os = "linux", target_env = "gnu")))] {
-            pub type PtraceRequest = ::libc::c_uint;
-        } else {
-            pub type PtraceRequest = c_int;
-        }
+
+cfg_if! {
+    if #[cfg(any(all(target_os = "linux", arch = "s390x"),
+                all(target_os = "linux", target_env = "gnu")))] {
+        pub type RequestType = ::libc::c_uint;
+    } else {
+        pub type RequestType = ::libc::c_int;
+    }
+}
+
+libc_enum!{
+    #[cfg_attr(all(any(all(target_os = "linux", arch = "s390x"),
+    all(target_os = "linux", target_env = "gnu"))), repr(u32))] 
+    #[cfg_attr(not(any(all(target_os = "linux", arch = "s390x"),
+    all(target_os = "linux", target_env = "gnu"))), repr(i32))] 
+    pub enum Request {
+        PTRACE_TRACEME, 
+        PTRACE_PEEKTEXT,
+        PTRACE_PEEKDATA,
+        PTRACE_PEEKUSER,
+        PTRACE_POKETEXT,
+        PTRACE_POKEDATA,
+        PTRACE_POKEUSER,
+        PTRACE_CONT,
+        PTRACE_KILL,
+        PTRACE_SINGLESTEP,
+        #[cfg(any(target_env = "musl", target_arch ="x86_64", target_arch = "s390x"))]
+        PTRACE_GETREGS,
+        #[cfg(any(target_env = "musl", target_arch ="x86_64", target_arch = "s390x"))]
+        PTRACE_SETREGS,
+        #[cfg(any(target_env = "musl", target_arch ="x86_64", target_arch = "s390x"))]
+        PTRACE_GETFPREGS,
+        #[cfg(any(target_env = "musl", target_arch ="x86_64", target_arch = "s390x"))]
+        PTRACE_SETFPREGS,
+        PTRACE_ATTACH,
+        PTRACE_DETACH,
+        #[cfg(any(target_env = "musl", target_arch ="x86_64", target_arch = "s390x"))]
+        PTRACE_GETFPXREGS,
+        #[cfg(any(target_env = "musl", target_arch ="x86_64", target_arch = "s390x"))]
+        PTRACE_SETFPXREGS,
+        PTRACE_SYSCALL,
+        PTRACE_SETOPTIONS,
+        PTRACE_GETEVENTMSG,
+        PTRACE_GETSIGINFO,
+        PTRACE_SETSIGINFO,
+        #[cfg(any(target_env = "musl", target_arch ="x86_64", target_arch = "s390x"))]
+        PTRACE_GETREGSET,
+        #[cfg(any(target_env = "musl", target_arch ="x86_64", target_arch = "s390x"))]
+        PTRACE_SETREGSET,
+        PTRACE_SEIZE,
+        PTRACE_INTERRUPT,
+        PTRACE_LISTEN,
+        PTRACE_PEEKSIGINFO,
+    }
+}
+      
+libc_enum!{
+    #[repr(i32)]
+    pub enum Event {
+        PTRACE_EVENT_FORK,
```

Comment:
Could you add docs for each variant here? They can be summarized from the man pages.